android/src/main/java/com/magicleap/magicscript/plane/ARPlaneDetectorBridge.kt

```diff
@@ -0,0 +1,191 @@
+package com.magicleap.magicscript.plane
+
+import com.facebook.react.bridge.*
+import com.google.ar.core.HitResult
+import com.google.ar.core.Plane
+import com.magicleap.magicscript.utils.NativeReactObjectsProvider
+import com.magicleap.magicscript.utils.ReactObjectsProvider
+import kotlin.math.abs
+
+class ARPlaneDetectorBridge private constructor() {
+    private var onUpdateListener: OnPlanesUpdated? = null
+    private var onAddedListener: OnPlanesAdded? = null
+    private var onRemovedListener: OnPlanesRemoved? = null
+    private var onTappedListener: OnPlaneTapped? = null
+    private var lastPlanes: List<Plane>? = null
+    private var isDetecting: Boolean = false
+    private var detectionConfiguration: List<Plane.Type>? = null
+    var objectsProvider: ReactObjectsProvider = NativeReactObjectsProvider.INSTANCE
+
+    companion object {
+        private val ID = "id"
+        private val CENTER = "center"
+        private val VERTICES = "vertices"
+        private val TYPE = "type"
+        private val ERROR = "error"
+        val PLANE_TYPE = "planeType"
+        val PLANE_TYPE_VERTICAL = "vertical"
+        val PLANE_TYPE_HORIZONTAL = "horizontal"
+        val INSTANCE = ARPlaneDetectorBridge()
+    }
+
+    fun mapPlanesToWritableMap(plane: Plane): WritableMap {
+        val planeMap = objectsProvider.createMap()
+        val vertices = objectsProvider.createArray()
+        val polygonList = plane.polygon.array().toList()
+        val centerPose = plane.centerPose
+        polygonList.chunked(2).forEach {
+            val point = centerPose.transformPoint(floatArrayOf(it[0], 0f, it[1]))
+            vertices.pushArray(objectsProvider.createArray().apply {
+                pushDouble(point[0].toDouble())
+                pushDouble(point[1].toDouble())
+                pushDouble(point[2].toDouble())
+            })
+        }
+        planeMap.putString(TYPE, plane.type.name)
+        planeMap.putArray(VERTICES, vertices)
+        planeMap.putArray(CENTER, objectsProvider.createArray().apply {
+            pushDouble(plane.centerPose.tx().toDouble())
+            pushDouble(plane.centerPose.ty().toDouble())
+            pushDouble(plane.centerPose.tz().toDouble())
+        })
+        planeMap.putString(ID, abs(plane.hashCode()).toString())
+        return planeMap
+    }
+
+    fun mapToError(errorMessage: String): WritableMap {
+        return objectsProvider.createMap().apply {
+            putString(ERROR, errorMessage)
+        }
+    }
+
+    fun onPlaneUpdate(planes: List<Plane>) {
+        if (planes.isNotEmpty()) {
+            val filteredPlanes = planes.filter { detectionConfiguration?.contains(it.type) ?: true }
```

Comment:
`?: true` should really be true by default?

Reply:
Yes, if detectionConfiguration is null we should return all of the planes